virtcontainers/api.go

```diff
@@ -650,3 +650,7 @@ func ProcessListContainer(sandboxID, containerID string, options ProcessListOpti
 
 	return c.processList(options)
 }
+
+func ContainersMap() (map[string]string, error) {
```

Comment:
IIUC, the purpose is to find out a container's sandboxID. Can you simply the API to just do the job? e.g.,
```func CantainerSandbox(containerID string) (sandboxID string, err error)```
So that it does not make internal data structure part of the API.

Reply:
Well the answer is that we don't want virtcontainers to run under the same assumptions as the CLI. In this case that a container ID is unique across sandboxes, meaning there is a single sandbox ID corresponding to the container ID provided. That being said, the whole logic handling a map assumes so... I guess this could all be implemented into the CLI instead, but it would mean that we would have to store this somewhere in the filesystem. Where would it be ?
The nice thing about virtcontainers is that it was handling this for us completely.

Reply:
Yup, the use of a map does not look accurate even inside virtcontainers.

I would vote for implementing it in the cli and we can just store the sandboxID of a container in its own directory. Thus no global maps nor global on disk locks.

We can store it in the root storage of kata cli (default to `/var/run/kata-containers`), such as `/var/run/kata-containers/containers/<container_id>/state-or-whatever.json`.

Reply:
Well I have thought about it and I am gonna implement it in virtcontainers in a generic way, by using a hash table of containers pointing to a list of pods. Now, the function will return the list of pods related to a container ID, which is okay and compliant with virtcontainers. This way we keep the storage being handled by virtcontainers. From the CLI perspective, we will get a list of pods which should never contain more than one element.

Reply:
Well, in that case please make it optional so that we can opt out in the CRI use cases. We will definitely disable it in frakti. We might also want to disable it for cri-o and cri-containerd as well to remove the unnecessary global locking and extra file system access, once we decouple the spec validation of cri containers from docker containers. In the cri shim case, we only need global uniqueness of sandboxID and sandbox-wide uniqueness of containerID, which can both be validated in virtcontainers. Then the new API is only necessary for docker.

Reply:
What's more, we always use containerID as sandboxID in the docker use case. The uniqueness of containerID == the uniqueness of sandboxID. So I do not think we need the API at all. Just validate proper uniqueness of sandboxID and containerID in virtcontainers.

Reply:
@bergwolf I have introduced a flag in the SandboxConfig. It'll be up to Frakti to define that it needs a stateless configuration.